src/main/java/jnr/posix/BaseNativePOSIX.java

```diff
@@ -755,6 +755,22 @@ public int rename(CharSequence oldName, CharSequence newName) {
         return libc().rename(oldName, newName);
     }
 
+    public String gethostname() {
+        ByteBuffer buffer = ByteBuffer.allocate(256);
+        int result;
+        try {
+            result = libc().gethostname(buffer, buffer.capacity() - 1);
+        } catch (java.lang.UnsatisfiedLinkError e) {
+            result = -1;
+        }
+        if (result == -1) return helper.gethostname();
+        buffer.position(0);
+        while (buffer.hasRemaining() && buffer.get() != 0);
+        buffer.limit(buffer.position() - 1);
+        buffer.position(0);
+        return Charset.forName("US-ASCII").decode(buffer).toString();
```

Comment:
I am somewhat amazed we this is a clean 7bit set of chars on linux but the closest I could see that it wasn't is GNU gethostname will get name from uname() and that is pretty vague on not supporting more than 8bit ascii.   If someone can tell us how this breaks we will deal with it then.